projects/rocprofiler-systems/source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

```diff
@@ -906,16 +990,224 @@ get_kernel_dispatch_timestamps()
+    if(record.kind == ROCPROFILER_CALLBACK_TRACING_OMPT)
+    {
+        check_name = ompt_get_detailed_callback_name(record);
+        _name      = check_name;
+    }
+    if(_name.empty())
+        _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    else
+    _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    endif
+
+#    if(!(ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES > 0))
+    // Forces omp_parallel begin and end to have same name, allowing perfetto track to
+    // connect. This will be changed in the future
+    if(record.operation == ROCPROFILER_OMPT_ID_parallel_begin) _name = "omp_parallel";
+#    endif
+
+    if(get_use_timemory())
+    {
+        component::category_region<category::rocm_marker_api>::start<quirk::timemory>(
+            _name);
+    }
+
+    if(get_use_perfetto() && (record.kind == ROCPROFILER_CALLBACK_TRACING_OMPT))
+    {
+        auto args = callback_arg_array_t{};
+        if(config::get_perfetto_annotations())
+        {
+            rocprofiler_iterate_callback_tracing_kind_operation_args(record, save_args, 1,
+                                                                     &args);
+        }
+
+        uint64_t _beg_ts   = ts;
+        auto     stream_id = stream_id_top();
+
+        tracing::push_perfetto_ts(
+            category::rocm_ompt_api{}, _name.data(), _beg_ts,
+            ::perfetto::Flow::ProcessScoped(record.correlation_id.internal),
+            [&](::perfetto::EventContext ctx) {
+                if(config::get_perfetto_annotations())
+                {
+                    tracing::add_perfetto_annotation(ctx, "begin_ns", _beg_ts);
+                    tracing::add_perfetto_annotation(ctx, "corr_id",
+                                                     record.correlation_id.internal);
+                    if(stream_id.handle != 0)
+                        tracing::add_perfetto_annotation(ctx, "stream_id",
+                                                         stream_id.handle);
+                    for(const auto& [key, val] : args)
+                    {
+                        tracing::add_perfetto_annotation(ctx, key, val);
+                    }
+                }
+            });
+    }
+}
+
+void
+ompt_tracing_callback_stop(
+    rocprofiler_callback_tracing_record_t record, rocprofiler_user_data_t* /*user_data*/,
+    rocprofiler_timestamp_t               ts,
+    std::optional<std::vector<tim::unwind::processed_entry>>& _bt_data)
+{
+    std::string_view _name;
+#    if(ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES > 0)
+    std::string check_name;
+    if(record.kind == ROCPROFILER_CALLBACK_TRACING_OMPT)
+    {
+        check_name = ompt_get_detailed_callback_name(record);
+        _name      = check_name;
+    }
+    if(_name.empty())
+        _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    else
+    _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    endif
```

Comment:
```suggestion
    std::string_view _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
```
Let's keep these "ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES" changes on a feature branch, for now.